src/backend/aspen/workflows/nextclade/prep_samples.py

```diff
@@ -0,0 +1,119 @@
+"""
+Writes out a FASTA for the sample PK ids specified by file.
+
+Core approach here is a copy of what's in workflows/pangolin/export.py.
+"""
+import io
+import json
+from typing import IO, Iterable, Optional
+
+import click
+from sqlalchemy.orm import joinedload
+
+from aspen.config.config import Config
+from aspen.database.connection import (
+    get_db_uri,
+    init_db,
+    session_scope,
+    SqlAlchemyInterface,
+)
+from aspen.database.models import Pathogen, Sample, UploadedPathogenGenome
+
+
+@click.command("export")
+@click.option("sample_ids_fh", "--sample-ids-file", type=click.File("r"), required=True)
+@click.option("sequences_fh", "--sequences", type=click.File("w"), required=True)
+@click.option(
+    "pathogen_info_fh", "--pathogen-info-file", type=click.File("w"), required=True
+)
+def cli(
+    sample_ids_fh: io.TextIOBase,
+    sequences_fh: io.TextIOBase,
+    pathogen_info_fh: IO[str],
+):
+    """
+    Writes out a FASTA for the specified samples.
+
+    - sample_ids_fh: A file of sample ID primary keys, one ID per line.
+        Each sample must be for the same pathogen (all SARS-CoV-2, etc)
```

Comment:
yeah, but why bother having a separate batch-creation script? This script can handle that functionality with a few minor adjustments.

Reply:
Wouldn't it need to do a mixed batch of pathogens then? The batch-creation script is intended to handle all pathogens at once on a scheduled basis. I figured a good dividing line was for the on-demand to handle a single pathogen at a time.

Reply:
The scheduled job can just kick off one sub-job per pathogen, it doesn't have to create the batches itself.

Reply:
Talked on Zoom about this: I'm sold! We'll use the same process for the sub-job and it will use a lack of sample_ids (or maybe another arg?) as a way to indicate a general run for the pathgoen. Then also adding a little additional logic for checking over the entire set of samples for a given pathogen and updating the stales in that case.

For now, I'll leave this PR as-is around this aspect, but I'll plan to use that approach when I add in the scheduled aspect down the road.

Reply:
I think the sub-job (and tbh, even this on-demand job) would benefit from getting an explicit pathogen argument.

Reply:
Sure, but I still think we should make use of the implicit pathogen in the listed samples, even if it's just to raise an error when they differ.

Reply:
fair enough.